### DISCUSSION TITLE
Hey! pdf compression in imagick

### DISCUSSION THREAD

**Discussion by waqaspuri** [ImageMagick/ImageMagick#2288] [Help]
Am not too late to ask this question from this forum. but anyways i find there is no solution yet. might be but I didn't get it. so let come to the point. I have more than 8 pages pages (it's like a newspaper each page) and it is in jpg format, and per php script below am making a pdf file out of it. like combined.pdf files. but I've a trouble **reducing the overall file size of pdf. for instance. the file size some time is much higher sometime goes higher above 40 to 45mbs.** 

so, my question to the point. how to reduce the pdf quality. i tried image compression but doesn't seems to be working in this case. 

`if($_GET['pdf'] == "do") {
					
					//echo "date generated ... ~ ! ";
				
				$pdf = new Imagick($pdf_page);	
				$compression_type = Imagick::COMPRESSION_JPEG;

				$pdf->setImageCompression($compression_type);
				$pdf->setImageCompressionQuality(20); 
				
				$pdf->setImageFormat('pdf');
			     
							 
				$pdf->writeImages('combined.pdf', true);	
				$pdf->clear();
				$pdf->destroy();
			
				
				
				
				ob_end_clean(); 
				 

				
				header('Content-type: application/pdf');
				header('Content-Disposition: attachment; filename="combined.pdf"');
				
				
				
				readfile('combined.pdf');`

**the array collects the pages, and create the combined.pdf. but how do I compress a pdf file?**

**Comment by snibgo** [comment]
What version of ImageMagick?

> ... but I've a trouble reducing the overall file size of pdf. for instance. the file size some time is much higher sometime goes higher above 40 to 45mbs.

You are reading some JPEG images, and writing those to a PDF, with JPEG compression. Is that correct?

Then the PDF will contain embedded raster JPEG images. It will be the sum of the size of those images, plus a small overhead.

If that doesn't help, please demonstrate the problem with a command-line example.

**Reply by waqaspuri** [reply]
the module version is 3.4.3. 
I don't understand what do you mean by "If that doesn't help".  did you mention something to me? @snibgo

**Reply by waqaspuri** [reply]
You are reading some JPEG images, and writing those to a PDF, with JPEG compression. Is that correct?
**Yes correct. in a combined pdf format.**

**Comment by snibgo** [comment]
A PDF file containing JPEG images will have a total size of those images, plus a small óverhead. If that isn't the case for you, please post a sample PDF with the problem.

**Comment by mattsrinc** [comment]
You should be able to compress the PDF with using either Group4 or ZIP compression:

[https://legacy.imagemagick.org/Usage/formats/#pdf_options](https://legacy.imagemagick.org/Usage/formats/#pdf_options)

ZIP should shrink the PDF down, maybe to one third of the original size.

How to use it with IMagick? Add this to your script and report:

`$pdf->setCompression(Imagick::COMPRESSION_ZIP);`
